### PR TITLE
Add grpc support

### PR DESCRIPTION
**Pull request by spuder**
Resolves issue https://github.com/solarkennedy/puppet-consul/issues/498

### CODE REVIEW

File: manifests/check.pp [hunk 1]
```diff
@@ -64,6 +67,7 @@
   $timeout    = undef,
   $token      = undef,
   $ttl        = undef,
+  $grpc       = undef,
```
Comment (bastelfreak) [inline manifests/check.pp]: can you please add a suitable datatype for the new parameter?

**Comment by bastelfreak** [comment]
Hi @spuder, thanks for the PR. can you please rebase after https://github.com/solarkennedy/puppet-consul/pull/500 got solved?

File: manifests/check.pp (hunk 1 above)
Comment (spuder) [reply manifests/check.pp]: You mean like this?
```
  $token      = undef,
  $ttl            = undef,
  String[1]     $grpc       = undef,
```

Comment (spuder) [reply manifests/check.pp]: Something like this? 

```
  $token = undef,
  $ttl = undef,
  String[1] $grpc = undef,
```

Comment (bastelfreak) [reply manifests/check.pp]: yes :)

File: manifests/check.pp [hunk 2]
```diff
@@ -64,6 +67,7 @@
   $timeout    = undef,
   $token      = undef,
   $ttl        = undef,
+  String[1] $grpc       = undef,
```
Comment (bastelfreak) [inline manifests/check.pp]: please adjust the alignment for all the `=` chars.

**Comment by solarkennedy** [comment]
master "should" be green now. rebase on master at will please.

**Comment by spuder** [comment]
Rebased

**Comment by bastelfreak** [comment]
@spuder can you take a look at the failing travis tests?

**Comment by spuder** [comment]
I've rebased and pushed again. The failing CI tests appear to be unrelated to any changes that I have made. I'm not quite sure why they are broken since it is referencing a file that doesn't even exist `spec/fixtures/modules/consul/manifests/params.pp`

```
     # Puppet::PreformattedError:
     #   Evaluation Error: Operator '[]' is not applicable to an Undef Value. (file: /home/travis/build/solarkennedy/puppet-consul/spec/fixtures/modules/consul/manifests/params.pp, line: 18, column: 17)
     #   ./vendor/bundle/ruby/2.4.0/gems/puppet-5.5.18/lib/puppet/pops/evaluator/runtime3_support.rb:39:in `optionally_fail'
  2) consul::watch on opensuse-13-x86_64  with valid type and handler is expected to contain File[/etc/consul/watch_my_watch.json] with content =~ /"handler" *: *"handler_path"/ and content =~ /"type" *: *"nodes"/
     Failure/Error:
       should contain_file('/etc/consul/watch_my_watch.json') \
           .with_content(/"handler" *: *"handler_path"/) \
           .with_content(/"type" *: *"nodes"/)
     Puppet::PreformattedError:
       Evaluation Error: Error while evaluating a Resource Statement, Evaluation Error: Operator '[]' is not applicable to an Undef Value. (file: /home/travis/build/solarkennedy/puppet-consul/spec/fixtures/modules/consul/manifests/params.pp, line: 18, column: 17) (line: 2) on node travis-job-d47f9421-2068-400d-a553-40e0d674f975.c.travis-ci-prod-3.internal
```

**Pull request closed without merging**